pkg/machine/ignition.go

```diff
@@ -208,12 +174,54 @@ WantedBy=sysinit.target
 				Name:     "remove-moby.service",
 				Contents: &deMoby,
 			},
-			{
-				Enabled:  boolToPtr(true),
-				Name:     "envset-fwcfg.service",
-				Contents: &envset,
-			},
 		}}
+
+	if len(vmtype) > 1 {
+		return errors.Errorf("got a lot of vmtype, has to be 0 or 1")
+	}
+	// QEMU specific part
+	if len(vmtype) == 1 && vmtype[0] == "qemu" {
```

Comment:
Similarly here — the provider could supply a `consumeQemuFWCfg` boolean, or something like that, instead of an untransparent and undocumented behavior difference based on a name.

Forcing the user to explicitly supply a value would also be simpler than worrying about `len(vmtype) != 1`.

Reply:
Thanks. Fixed